src/System.Windows.Forms/src/System/Windows/Forms/ComponentManagerBroker.cs

```diff
@@ -475,249 +437,214 @@ bool UnsafeNativeMethods.IMsoComponentManager.FRegisterComponent(UnsafeNativeMet
         {
-            int dwLocalComponentID = unchecked((int)(long)dwComponentID);
-            if (_original == null)
-            {
-                return false;
-            }
-
-            if (_components == null || dwLocalComponentID <= 0 || !_components.ContainsKey(dwLocalComponentID))
+            if (_original == null || _components == null || !_components.ContainsKey(*dwComponentID))
             {
-                return false;
+                return BOOL.FALSE;
             }
 
             if (_refCount == 1 && SafeNativeMethods.GetCurrentThreadId() == _creationThread)
             {
                 if (!RevokeComponent())
                 {
-                    return false;
+                    return BOOL.FALSE;
                 }
             }
 
             _refCount--;
-            _components.Remove(dwLocalComponentID);
+            _components.Remove(*dwComponentID);
 
             Debug.Assert(_refCount >= 0, "underflow on ref count");
             if (_refCount <= 0)
             {
                 Dispose();
             }
 
-            if (dwLocalComponentID == _activeComponentId)
+            if (*dwComponentID == _activeComponentId)
             {
                 _activeComponent = null;
                 _activeComponentId = 0;
             }
-            if (dwLocalComponentID == _trackingComponentId)
+            if (*dwComponentID == _trackingComponentId)
             {
                 _trackingComponent = null;
                 _trackingComponentId = 0;
             }
 
-            return true;
+            return BOOL.TRUE;
         }
 
-        bool UnsafeNativeMethods.IMsoComponentManager.FUpdateComponentRegistration(IntPtr dwComponentID, NativeMethods.MSOCRINFOSTRUCT info)
-        {
-            if (_original == null)
-            {
-                return false;
-            }
-            // We assume that all winforms domains use the same registration.
-            return _original.FUpdateComponentRegistration(_componentId, info);
-        }
+        unsafe BOOL IMsoComponentManager.FUpdateComponentRegistration(
+            uint* dwComponentID,
+            MSOCRINFO* pcrinfo)
+            => _original?.FUpdateComponentRegistration(dwComponentID, pcrinfo) ?? BOOL.FALSE;
 
-        bool UnsafeNativeMethods.IMsoComponentManager.FOnComponentActivate(IntPtr dwComponentID)
+        unsafe BOOL IMsoComponentManager.FOnComponentActivate(uint* dwComponentID)
         {
-            int dwLocalComponentID = unchecked((int)(long)dwComponentID);
-            if (_original == null)
-            {
-                return false;
-            }
             // Activation requres us to store the currently active component.  We will send data to it
-            if (_components == null || dwLocalComponentID <= 0 || !_components.ContainsKey(dwLocalComponentID))
+            if (_original == null || _components == null || !_components.ContainsKey(*dwComponentID)
```

Comment:
Don't dereference the dwComponentID, the original source as correct. The headers have a DWORD_PTR which is a pointer sized opaque value. You are not supposed to dereference.
```suggestion
            if (_original == null || _components == null || !_components.ContainsKey(dwComponentID)
```

Reply:
It should be `UIntPtr*` as far as I can tell.

Reply:
nvm